keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -50,103 +44,86 @@ class RandomResizedCrop(BaseImageAugmentationLayer):
                                   bounding_box=None):
-        # random area and aspect ratio
-        random_area = (1.0 +
-                       preprocessing.random_inversion(self._random_generator) *
-                       self.area_factor())
-        random_aspect_ratio = (
-            1.0 + preprocessing.random_inversion(self._random_generator) *
-            self.aspect_ratio_factor())
-
-        # corresponding height and width (1 = original height/width)
-        new_height = tf.sqrt(random_area / random_aspect_ratio)
-        new_width = tf.sqrt(random_area * random_aspect_ratio)
-
-        # random offsets for the crop, inside or outside the image
-        height_offset = self._random_generator.random_uniform(
+        area_factor = self.area_factor()
+        aspect_ratio = tf.random.uniform((), minval=self.aspect_ratio_factor[0], 
+            maxval=self.aspect_ratio_factor[1], dtype=tf.float32)
+
+        new_height = tf.clip_by_value(tf.sqrt(area_factor / aspect_ratio), 0.0,
+                                    1.0)  # to avoid unwanted/unintuitive effects
+        new_width = tf.clip_by_value(tf.sqrt(area_factor * aspect_ratio), 0.0, 1.0)
+
+        height_offset = tf.random.uniform(
             (),
-            tf.minimum(0.0, 1.0 - new_height),
-            tf.maximum(0.0, 1.0 - new_height),
+            minval=tf.minimum(0.0, 1.0 - new_height),
+            maxval=tf.maximum(0.0, 1.0 - new_height),
             dtype=tf.float32,
         )
-        width_offset = self._random_generator.random_uniform(
+
+        width_offset = tf.random.uniform(
             (),
-            tf.minimum(0.0, 1.0 - new_width),
-            tf.maximum(0.0, 1.0 - new_width),
+            minval=tf.minimum(0.0, 1.0 - new_width),
+            maxval=tf.maximum(0.0, 1.0 - new_width),
             dtype=tf.float32,
         )
 
-        return (new_height, new_width, height_offset, width_offset)
+        y1 = height_offset
+        y2 = height_offset + new_height
+        x1 = width_offset
+        x2 = width_offset + new_width
+
+        return [[y1, x1, y2, x2]]
 
     def augment_image(self, image, transformation):
-        height, width, _ = image.shape
         image = tf.expand_dims(image, axis=0)
+        boxes = transformation
 
-        new_height, new_width, height_offset, width_offset = transformation
-
-        transform = RandomResizedCrop._format_transform([
-            new_width,
-            0.0,
-            width_offset * width,
-            0.0,
-            new_height,
-            height_offset * height,
-            0.0,
-            0.0,
-        ])
-        image = preprocessing.transform(
-            images=image,
-            transforms=transform,
-            interpolation=self.interpolation,
-            fill_mode=self.fill_mode,
-            fill_value=self.fill_value,
+        # See bit.ly/tf_crop_resize for more details
+        augmented_image = tf.image.crop_and_resize(
```

Comment:
nice use of tf.image - but you won't be using fill mode if you do this.  Note that our `preprocessing.transform` function does actually support mirror fill modes, while tf.image.crop_and_resize() will not.